pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -106,20 +107,14 @@ def config(self) -> NoiseModel:
         return self._config
 
     def _build_collapse_operators(self, config: NoiseModel) -> None:
-        def basis_check(noise_type: str) -> None:
-            """Checks if the basis allows for the use of noise."""
-            if self.basis_name == "all":
-                # Go back to previous config
-                raise NotImplementedError(
-                    f"Cannot include {noise_type} noise in all-basis."
-                )
 
         local_collapse_ops = []
         if "dephasing" in config.noise_types:
             dephasing_rates = {
                 "d": config.dephasing_rate,
                 "r": config.dephasing_rate,
                 "h": config.hyperfine_dephasing_rate,
+                # TODO: Add dephasing rate for leakage
```

Comment:
Truth is I don't know... If you only have dephasing noise, then I guess it does not make sense since you don't expect "x" to be populated at any time ?